src/phoenix/trace/openai/instrumentor.py

```diff
@@ -0,0 +1,258 @@
+        tracer (Tracer): The tracer to use to create spans.
+
+    Returns:
+        Callable[..., Any]: The wrapped request method.
+    """
+
+    def wrapped(*args: Any, **kwargs: Any) -> Any:
+        call_signature = signature(request_fn)
+        bound_arguments = call_signature.bind(*args, **kwargs)
+        parameters = bound_arguments.arguments["params"]
+        is_streaming = parameters.get("stream", False)
+        url = bound_arguments.arguments["url"]
+        current_status_code = SpanStatusCode.UNSET
+        events: List[SpanEvent] = []
+        attributes: SpanAttributes = dict()
+        if not is_streaming and _get_request_type(url) is RequestType.CHAT_COMPLETION:
+            for (
+                attribute_name,
+                get_parameter_attribute_fn,
+            ) in _PARAMETER_ATTRIBUTE_FUNCTIONS.items():
+                if attribute_value := get_parameter_attribute_fn(parameters):
+                    attributes[attribute_name] = attribute_value
+            outputs = None
+            try:
+                start_time = datetime.now()
+                outputs = request_fn(*args, **kwargs)
+                end_time = datetime.now()
+                current_status_code = SpanStatusCode.OK
+                return outputs
+            except Exception as error:
+                end_time = datetime.now()
+                current_status_code = SpanStatusCode.ERROR
+                events.append(
+                    SpanException(
+                        message=str(error),
+                        timestamp=end_time,
+                        exception_type=type(error).__name__,
+                        exception_stacktrace=get_stacktrace(error),
+                    )
+                )
+                raise
+            finally:
+                if outputs:
+                    response = outputs[0]
+                    for (
+                        attribute_name,
+                        get_response_attribute_fn,
+                    ) in _RESPONSE_ATTRIBUTE_FUNCTIONS.items():
+                        if attribute_value := get_response_attribute_fn(response):
+                            attributes[attribute_name] = attribute_value
+                tracer.create_span(
+                    name="openai.ChatCompletion.create",
+                    span_kind=SpanKind.LLM,
+                    start_time=start_time,
+                    end_time=end_time,
+                    status_code=current_status_code,
+                    status_message="",
+                    attributes=attributes,
+                    events=events,
+                )
+        else:
+            return request_fn(*args, **kwargs)
+
+    return wrapped
+
+
+def _input_value(parameters: Parameters) -> Optional[str]:
+    return json.dumps(messages) if (messages := parameters.get("messages")) else None
+
+
+def _input_mime_type(parameters: Parameters) -> MimeType:
+    return MimeType.JSON
```

Comment:
this function just returns a constant, maybe it doesn't need to be a function?

Reply:
Probably easiest for it to be a function, otherwise, we'd have to check for callables vs. non-callables.

Reply:
addressed in 95e3ae92